generator/Parser/Tasks/ParseProperty.php

```diff
@@ -3,37 +3,35 @@
 namespace Spatie\SchemaOrg\Generator\Parser\Tasks;
 
 use Spatie\SchemaOrg\Generator\Property;
-use Symfony\Component\DomCrawler\Crawler;
 
 class ParseProperty extends Task
 {
     public function __invoke(): ?Property
     {
-        $node = new Crawler($this->definition);
-
         $property = new Property();
 
-        $property->name = $this->getText($node, '[property="rdfs:label"]');
+        $property->name = $this->getDefinitionProperty('rdfs:label');
 
         if (empty($property->name)) {
             return null;
         }
 
-        $property->description = $this->getText($node, '[property="rdfs:comment"]');
+        $property->description = $this->getDefinitionProperty('rdfs:comment');
 
-        $property->resource = $this->getAttribute($node, 'resource');
+        $property->resource = $this->getResource();
 
-        $node
-            ->filter('[property="http://schema.org/domainIncludes"]')
-            ->each(function (Crawler $domain) use ($property) {
-                $property->addType($this->getText($domain));
+        $this->getWrappedDefinitionProperty('http://schema.org/domainIncludes')
+            ->each(function (array $domain) use ($property) {
+                $property->addType($this->getResourceName($domain));
             });
 
-        $node
-            ->filter('[property="http://schema.org/rangeIncludes"]')
-            ->each(function (Crawler $range) use ($property) {
+        $this->getWrappedDefinitionProperty('http://schema.org/rangeIncludes')
+            ->filter(function ($value) {
+                return ! in_array($this->getResourceName($value), ['GeospatialGeometry', 'EducationalOccupationalCredential', 'DefinedTerm', 'VirtualLocation', 'PhysicalActivityCategory', 'GovernmentBenefitsType'], true);
+            })
```

Comment:
Rather than flagging types as pending, this filter is excluding them from being added to the domain and range includes. The main effective difference is that it prevents certain functions from being downgraded to `@method` doc blocks.

With this filter removed - and not adding any logic to the TypeCollection yet - every Schema with a `location` method will be affected.

Reply:
So I think I didn't quite explain what I was getting at well and still don't think I can. Essentially what I was trying to solve was removing pending Types from a property, instead of marking a property as pending later.

The main difference in the affect is that doing it this way allows a property to be generated as published without a pending type as an option. Yet, by removing this the existing code marking the property as pending works as intended, but due to the difference between RDFa and JSON-LD content affects the generated PHP code.

Interestingly this all may be inconsequential in the grand-scheme, since from my other testing with future versions from schemaorg things change allowing this to be removed. From what I found testing with the 9.0 schema they do a better job about keeping the released file internally consistent.

There's only one caveat, the "Pending" types now have two meanings in the 9.0 release. There are truly pending types and properties that are not included in the equivalent release file, and then there are what I consider "Beta Pending" items as they are published in the main file yet still marked as pending.

Reply:
Ok - so I have implemented an effectively identical solution without the hardcoded types. So now it filters things cleanly when it is parsing other similar elements.